### DISCUSSION TITLE
6.3.12 -> 6.4.0 breaks storybook

### DISCUSSION THREAD

**Discussion by forgo** [storybookjs/storybook#16816] [Help]
I'm posting this in Q&A rather than as a bug because I can't easily recreate this issue (my company's project is private). I'm just hoping for some general guidance/ideas here.

I recently rehauled an entire React/Typescript component library to use Storybook `6.3.12`. I ended up mostly nuking a lot of configs and starting from absolute scratch because I found the migration process to be an absolute nightmare. Suffice to say, I got things back and running with all our components showing up again. And the performance was much better!

However, I noticed there was another big release a few days ago `6.4.0`, and I attempted first what the console suggested:

`npx sb@latest upgrade`

This prompted me to add the following to my `main.js`:
` framework: "@storybook/react"` which I confirmed with `y`

Then prompted me to use the eslint plugin designed for storybook, (which I again confirmed with a `y`). This ended up modifying my existing `eslintrc.cjs`.

However this resulted in a series of warnings and errors and prevented storybook completely from buildling. 

First, it seemed to stop understanding typescript in any of my story `.tsx` files, as the first line using a typescript keyword would complain with a syntax error such as:

`Missing semicolon.`

I 1st attempted to revert all these upgrades and re-ran the upgrade but NOT adding the eslint plugin. This failed in the same way.

My 2nd attempt was to comment out the new framework field `  // framework: "@storybook/react"`. This failed in the same way.

Third, I thought maybe `npx` was a bad idea, seeing as how I'm using yarn berry, and so I attempted to revert and re-try with:

`yarn dlx sb@latest upgrade`

This also failed.

Disregarding all the `Missing semicolon` errors. Some messages at the end of the storybook output (before it exits), are also confusing:

```
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      request is not managed by the pnpapi
        es-get-iterator tried to access process. While this module is usually interpreted as a Node builtin, your resolver is running inside a non-Node resolution context where such builtins are ignored. Since process isn't otherwise declared in es-get-iterator's dependencies, this makes the require call ambiguous and unsound.
        Required package: process
        Required by: /Users/elliott/IdeaProjects/react-components/.yarn/cache/es-get-iterator-npm-1.1.2-e8db4dc1b9-f75e66acb6.zip/node_modules/es-get-iterator/
```

Can anyone help me find out what I'm missing here?  I'd like to make sure our project is up-to-date as possible so that any potential v7 upgrades will be more seamless.

I was only a few days out of date on the version, and now the next upgrade (6.4.0) is introducing all sorts of headaches again with the version bump and minor config modifications.

I tried changing my eslint config to be `eslintrc.js` as opposed to `eslintrc.cjs`, but nothing changed there.

**Comment by shilman** [comment]
Hi @forgo sorry to hear about the headaches upgrading. We tried to flush out as many issues as possible during the prerelease period for 6.4, but there are an infinite number of project configurations and it's still something we're struggling with.

I'd like to help you get this solved. Can you:
1. try skipping the eslint stuff entirely and see if that fixes any of the warnings
2. if you're using the `webpack5` builder, try commenting out the  following line for `process`- related error and re-running https://github.com/storybookjs/storybook/pull/16725/files#diff-0c5866c9abec79cf6cc2faa8f6b527072f8ee0206aa1fdc9f30736f2f7980ef7R217 The relevant file will be `@storybook/builder-webpack5/dist/cjs/preview/iframe-webpack.config.js`

**Comment by forgo** [comment]
@shilman Thank you for the response. Sorry I haven't gotten back to this right away. I did try your suggestion 1. As far as your 2 suggestion, I cannot easily modify the third-party dependency code as I'm using yarn berry which stores these dependencies with a zero-install configuration and are zip files.

**Reply by shilman** [reply]
Hi @forgo, In that case we have a few options:

1. do nothing
2. you provide a public reproduction and I can take a look
3. try downgrading `@storybook/*` to `6.4.0-rc.3` which is the version before the change i'd like to try backing out

Thanks for your patience!

**Reply by forgo** [reply]
Hi @shilman, when I get a chance, I will try to downgrade. An interesting note is that I got this working with 6.3.x and pushed our updates to our main branch. When another developer rebased off of this, they started to see this same error (despite it being 6.3).

I only mention that because it seems to point to an upgrade cleanup path issue. Considering I first ran into this issue upgrading the project in place, and I had to nuke things and start from scratch to get rid of this error.

Furthermore, even fixing the issue, and having another developer merge with those changes, something in yarn was possibly being cached on their side. We could never really isolate exactly what was being cached though (we deleted yarn.lock, node_module, .yarn/cache, etc). I thought it could be the node_modules since I believe babel is putting things in there despite this being a yarn project, but the problem persisted.

The only way the other developer was able to get my updates using 6.3.x working was to entirely nuke and re-clone from our main branch. Perhaps that gives you some insight and narrows down what potentially could be going on here, but I get the feeling it's some particular yarn config file or auto-generated file that isn't obviously being updated in the Storybook upgrade path.

I hope that context helps for now.

**Comment by CWSites** [comment]
I was also on `6.3.12`, I updated to `6.4.0` and I'm able to start storybook, however as soon as I try to open one of the components it hangs and I don't get any useful errors. I also tried downgrading to `6.4.0-rc.3` and had the same issue.

These are the packages that I have and tried with both.
```
    "@storybook/addon-a11y": "6.4.0-rc.3",
    "@storybook/addon-controls": "6.4.0-rc.3",
    "@storybook/addon-essentials": "6.4.0-rc.3",
    "@storybook/addons": "^5.0.0",
    "@storybook/components": "6.4.0-rc.3",
    "@storybook/core-events": "6.4.0-rc.3",
    "@storybook/react": "6.4.0-rc.3",
    "@storybook/theming": "6.4.0-rc.3",
```

This is what I had before that worked fine.

```
    "@storybook/addon-a11y": "^6.3.8",
    "@storybook/addon-actions": "^5.2.0-beta.13",
    "@storybook/addon-controls": "^6.3.8",
    "@storybook/addon-essentials": "^6.3.8",
    "@storybook/addons": "^5.0.0",
    "@storybook/components": "^6.3.0",
    "@storybook/core-events": "^6.3.0",
    "@storybook/react": "^6.3.8",
    "@storybook/theming": "^6.0.0",
```

**Reply by shilman** [reply]
@CWSites All of your storybook package versions should match.

You can do this automatically (recommended) by:
- stable: `npx sb upgrade`
- prerelease: `npx sb upgrade --prerelease`

**Reply by CWSites** [reply]
Even when I had every single one of those matching at either `6.4.0-rc.3` or `6.4.0` I had the same issue.
